tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -721,7 +721,7 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
 		framework.ExpectNoError(err, "Expected TiDB cluster scaled out and ready")
 
 		err = controller.GuaranteedUpdate(genericCli, tc, func() error {
-			tc.Spec.Version = utilimage.TiDBV3Version
+			tc.Spec.Version = utilimage.TiDBV3UpgradeVersion
```

Comment:
this is a bug I made in a previous PR, In this upgrade case, we should use an new version